litmus-portal/graphql-server/pkg/myhub/myhub.go

```diff
@@ -292,9 +296,15 @@ func GetYAMLData(ctx context.Context, experimentInput model.ExperimentInput) (st
 	return YAMLData, nil
 }
 
-// GetPredefinedExperimentYAMLData is responsible for sending the experiment/engine.yaml for a given experiment.
+// GetPredefinedExperimentYAMLData is responsible for sending the workflow.yaml for a given pre-defined workflow.
 func GetPredefinedExperimentYAMLData(ctx context.Context, experimentInput model.ExperimentInput) (string, error) {
-	YAMLPath := handler.GetPredefinedExperimentManifest(ctx, experimentInput)
+	var YAMLPath string
+	if strings.ToLower(experimentInput.FileType.String()) != "workflow" {
+		return "", errors.New("invalid file type")
+	}
+	if strings.ToLower(experimentInput.ChartName) == "predefined" && strings.ToLower(experimentInput.FileType.String()) == "workflow" {
+		YAMLPath = handler.GetPredefinedExperimentManifest(ctx, experimentInput)
+	}
```

Comment:
The file type checks isn't good enough, if the file type can change how the data is stored/operated on then it'll be better to check the actual k8s object and verify the `Kind` of the resource being sent.

Reply:
@gdsoumya FileType is basically used to distinguish between different files from the chaos-charts repository and generate the hub path for the same. Not sure if we need another check to verify the `Kind` here.